src/app/component/user/components/profile/profile-cards/profile-cards.component.ts

```diff
@@ -1,23 +1,39 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { catchError } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+import { CardModel } from '@user-models/card.model';
 import { ProfileService } from '../profile-service/profile.service';
 
 @Component({
   selector: 'app-profile-cards',
   templateUrl: './profile-cards.component.html',
   styleUrls: ['./profile-cards.component.scss']
 })
-export class ProfileCardsComponent implements OnInit {
+export class ProfileCardsComponent implements OnInit, OnDestroy {
+  public profileSubscription: Subscription;
+  public factOfTheDay: CardModel;
+  public error;
 
-   constructor(private profileService: ProfileService) { }
-
-   public profileSubscription;
-   public facts = [];
+  constructor(private profileService: ProfileService) { }
 
   ngOnInit() {
-    this.profileSubscription = this.profileService.getFactsOfTheDay().subscribe(
-      (success) => {
-        this.facts = [success, ...this.facts];
-      }
-    );
+    this.getFactOfTheDay();
+  }
+
+  getFactOfTheDay(): void {
+    this.profileSubscription = this.profileService.getFactsOfTheDay()
+      .pipe(
+        catchError((error) => {
+          this.error = error;
+          return error;
+        })
+      )
+      .subscribe((success: CardModel) => {
+        this.factOfTheDay = success;
```

Comment:
is it object?
[https://dev.to/antonio_pangall/best-practices-to-keep-objects-and-arrays-immutable-in-javascript-3nmm](https://dev.to/antonio_pangall/best-practices-to-keep-objects-and-arrays-immutable-in-javascript-3nmm)

Reply:
yes it is object. fixed